src/Form/Type/OrderItemType.php

```diff
@@ -37,10 +36,13 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                     $this->orderItemQuantityModifier->modify($orderItem, $quantity);
                 },
             ])
-            // TODO: change to autocomplete type for product variant
-            ->add('variant', EntityType::class, [
-                'class' => ProductVariant::class,
-                'choice_label' => 'code',
+            ->add('variant', ResourceAutocompleteChoiceType::class, [
```

Comment:
Does this search in `id`, `code`, and `name` as stated in the issue?

Reply:
It searches by descriptor, which is code or name, as far as I know. Is the ID search a hard requirement? 😅 It's of course possible, but it would require from us to create a new repository function + new repository configuration and trait to implement it in the final application + new route. It's just easier this way and (I believe, correct me if I'm wrong) searching by name or code is much more useful than by id 🖖

Reply:
Okay. Let's just keep it this way for now :D